CMakeLists.txt

```diff
@@ -144,6 +144,16 @@ if(NOT CLANG_RESOURCE_DIR)
     endif()
     set(tried "${tried}\n  ${d}/clang/${v}")
   endforeach()
+  if(NOT CLANG_RESOURCE_DIR)
+    foreach(d ${LLVM_LIBRARY_DIRS})
+      string(REGEX REPLACE  "llvm.*" "" dd ${d})
+      if(IS_DIRECTORY "${dd}/clang/${v}/include")
+	set(CLANG_RESOURCE_DIR ${dd}/clang/${v})
+	break()
+      endif()
```

Comment:
Rather than duplicating the loop body, please prepare a list of candidate directories ahead of the above loop.